scripts/visual-active-effects.mjs

```diff
@@ -36,17 +36,18 @@ export class VisualActiveEffects extends Application {
       // Set up the various text (intro and content).
       const desc = entry.effect.flags["dfreds-convenient-effects"]?.description;
       const data = entry.effect.flags[MODULE]?.data ?? {};
+      const rollData = (await fromUuid(entry.effect.origin))?.getRollData() || {};
```

Comment:
I'm not sure that `ActiveEffect#origin` is the correct target to use to retrieve the roll data. This has proven to be quite unreliable. We should consider using simply the effect's parent instead (which may be an item or actor, but that is immaterial).

Reply:
That won't work in the use-case I was playing around with (a character has an aura that adds the effect to other characters).

Let's call the source Actor1 and the other party member Actor2.

When viewing Actor1's Visual Active Effects, everything is good.
When viewing Actor2's Visual Active Effects, the rollData refers to Actor2 (as the effect is on Actor2), so the details are incorrect. (screenshot attached of example, Actor2 is not a paladin, so is missing the scale value and has a different charisma modifier)

<img width="343" alt="image" src="https://github.com/krbz999/visual-active-effects/assets/293277/517a6782-8b1c-4070-807d-af0187e0f134">

Reply:
Do you have an example of how to trigger the unreliable origin value?

What about if we can't get the origin source, we then fall back to the effect's parent, and failing that, fall back to empty rollData?

Reply:
If we want to use `ActiveEffect#origin`, then we will have to consider these cases:
- the origin does not exist
- the origin is an actor
- the origin is an item
- the origin is an effect
- the origin for some reason points to something in a compendium

These are all cases that are easy to reproduce in the dnd5e system (before or after v3).

Reply:
Roger that. The 2nd and 3rd are fine (as the 3rd is the use case I’m testing - an item on a different actor), but I’ll see about recreating 4th and 5th. 5th I might need some pointers on how to trigger it. 

For the 1st point, I think falling back to the local parent makes sense (and should be the fallback for all of the cases.

Reply:
@krbz999 I've updated the code to handle the first 3 cases. I haven't been able to get a scenario where I can trigger cases 4 or 5.

I sent you a message on Patreon to try to get an example from you.

Reply:
Case 4 happens on dnd5e v3.0.x when an effect is transferred using the built-in functionality via the chatlog.

Case 5 can of course happens programmatically if a user forgets to change the `origin` property.

Reply:
I believe I've covered all of these cases now. It will use the origin if it can, but otherwise progressively fall back to the parent or an empty value.